test/evergreen/code_complexity_analysis.py

```diff
@@ -0,0 +1,137 @@
+import argparse
+import ast
+import csv
+import json
+import os.path
+import pandas as pd
+import sys
+
+from collections import defaultdict
+from heapq import nlargest
+
+def get_atlas_compatible_code_statistics(summaryFile, dataFile, outfile='atlas_code_complexity.json'):
+    ''' Generate the Atlas compatible format report. '''
+
+    atlasFormat = {
+                'Test Name': "Code Complexity",
+                'config': {},
+                'metrics': get_code_complexity(summaryFile, dataFile)
+            }
+
+    dirName = os.path.dirname(outfile)
+    if dirName:
+        os.makedirs(dirName, exist_ok=True)
+
+    with open(outfile, 'w') as outfile:
+        json.dump(atlasFormat, outfile, indent=4)
+
+def get_code_complexity(summaryFile, dataFile):
+    ''' Generate a list of intended contents from the complexity analysis. '''
+
+    global dataFrame
+    global topNRegions
+    dataFrame = pd.read_csv(dataFile)
+    topNRegions = 5
+    resultList = []
+    complexityDict = {}
+    complexityDict['Average'] = get_average(summaryFile)
+    complexityDict['Stats Ranges'] = get_complexity_ranges_list(dataFile)
+    complexityDict['Top ' + str(topNRegions) + ' Regions'] = get_region_list(dataFile)
+    
+    resultList.append(complexityDict)
+    return (resultList)
+
+def get_region_list(dataFile):
+    ''' Retrieve topN Regions/functions with the highest cyclomatic complexity values. '''
+
+    top5 = dataFrame.nlargest(topNRegions, 'std.code.complexity:cyclomatic')
+    atlasFormat = {}
+    for index, row in top5.iterrows():
+        atlasFormat[row["region"]] = row["std.code.complexity:cyclomatic"]
+
+    return (atlasFormat)
+
+def get_complexity_ranges_list(dataFile):
+    ''' Retrieve 3 set of cyclomatic complexity ranges, above 20, above 50 and above 90. '''
+
+    rangesList = [20, 50, 90]
+    atlasFormat = {}
+    for i in rangesList:
+        aboveRangeString = 'Above ' + str(i)
+        atlasFormat[aboveRangeString] = get_complexity_max_limit(i)
+
+    return (atlasFormat)
+
+def get_complexity_max_limit(max_limit: int):
+    ''' Retrieve the number of cyclomatic complexity values above the max_limit. '''
+
+    columnName = 'std.code.complexity:cyclomatic'
+    # Select column 'std.code.complexity:cyclomatic' from the dataframe
+    column = dataFrame[columnName]
+    # Get count of values greater than max_limit in the column 'std.code.complexity:cyclomatic'
+    count = column[column > max_limit].count()
+
+    # Pandas aggregation functions (like sum, count and mean) returns a NumPy int64 type number not a Python integer.
+    # Object of type int64 is not JSON serializable so need to convert into an int.
+
+    return (int(count))
+
+def get_average(complexity_summary_file):
+    ''' Retrieve the cyclomatic code complexity value '''
+    """
```

Comment:
The docstring markers """/''' are duplicated. One pair should be good enough.

Reply:
Fixed in 19d2a584b833bc89692c079e17033ea0c4609175